### PR TITLE
add startDrawing to public API [WIP]

### LINKED ISSUE #196: Expose startDrawing to the public api
Users should be able to opt out of our controls and still start drawing. This would be possible via a public version of `_startDrawing`. I think this should accept constant strings which we expose on Draw. Thoughts @kelvinabrokwa?

### LINKED ISSUE #216: Clicking on a draw button again before drawing anything will commit a premature feature
If you click on any of the draw buttons and then, instead of drawing anything, you click on a draw button again - a feature will now have been committed with a geometry with 0-coordinates. If you continue clicking on the draw buttons you will continue adding premature features.

### PR DESCRIPTION
**Pull request by kelvinabrokwa**
Closes #196
Closes #216

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -10,6 +10,12 @@ export default class API extends mapboxgl.Control {
 
   constructor() {
     super();
+    this.types = {
```
Comment (mcwhittemore) [inline src/api.js]: Should these be on this or should we have a constants object on the Draw constructor?

Comment (mcwhittemore) [inline src/api.js]: I don't know, many constants are a bit weird. People will likely just enter their own strings...

File: src/api.js [hunk 2]
```diff
@@ -105,6 +111,15 @@ export default class API extends mapboxgl.Control {
       });
   }
 
+  startDrawing(type) {
+    this._startDrawing(type);
+  }
+
+  _startDrawing() {
```
Comment (mcwhittemore) [inline src/api.js]: Lets drop private functions. What do we gain by making this kinda private?

File: src/api.js (hunk 1 above)
Comment (kelvinabrokwa) [inline src/api.js]: Constants will help us prevent bugs internally and externally. And I put them in API.js since its public-facing.

Comment (mcwhittemore) [inline src/api.js]: Sounds good.

**Comment by mcwhittemore** [comment]
Lets merge

**Comment by kelvinabrokwa** [comment]
@mcwhittemore :shipit:?

File: src/draw.js [hunk 3]
```diff
@@ -151,20 +151,20 @@ export default class Draw extends API {
...
         break;
-      case 'square':
+      case this.types.SQUARE:
         obj = new Square({ map: this._map });
         break;
-      case 'point':
+      case this.types.POINT:
         obj = new Point({ map: this._map });
         break;
       default:
```
Comment (mcwhittemore) [inline src/draw.js]: Should this through an error?

**Comment by mcwhittemore** [comment]
- [ ] add debug implementation (more buttons that start the drawing)
- [ ] Can we test this?

**Comment by mcwhittemore** [comment]
Ok. This LGTM. We still need to find write tests, but I think we should do this after we land a way to test draw without using gl. I'm pretty sure we can do this with a simple shim of mapbox-gl-js.

**Pull request merged**